netty/src/test/java/io/grpc/transport/netty/NettyClientTransportTest.java

```diff
@@ -118,45 +128,87 @@ public void teardown() throws Exception {
     group.shutdownGracefully(0, 10, TimeUnit.SECONDS);
   }
 
+  @Test
+  public void headersShouldAddDefaultUserAgent() throws Exception {
+    NettyClientTransport transport = createTransport();
+    transport.start(clientTransportListener);
+
+    // Send a single RPC and wait for the response.
+    unaryCall(transport, new Metadata.Headers()).get(10, TimeUnit.SECONDS);
+
+    // Verify that the received headers contained the User-Agent.
+    assertEquals(1, serverListener.streamListeners.size());
+
+    Metadata.Headers headers = serverListener.streamListeners.get(0).headers;
+    assertEquals(Utils.DEFAULT_USER_AGENT.toString(UTF_8), headers.get(HttpUtil.USER_AGENT));
+  }
+
+  @Test
+  public void headersShouldOverrideDefaultUserAgent() throws Exception {
+    NettyClientTransport transport = createTransport();
+    transport.start(clientTransportListener);
+
+    // Send a single RPC and wait for the response.
+    String userAgent = "testUserAgent";
+    Metadata.Headers sentHeaders = new Metadata.Headers();
+    sentHeaders.put(HttpUtil.USER_AGENT, userAgent);
+    unaryCall(transport, sentHeaders).get(10, TimeUnit.SECONDS);
+
+    // Verify that the received headers contained the User-Agent.
+    assertEquals(1, serverListener.streamListeners.size());
+    Metadata.Key<String> userAgentKey = Metadata.Key.of(Utils.USER_AGENT.toString(UTF_8),
+            Metadata.ASCII_STRING_MARSHALLER);
+    Metadata.Headers receivedHeaders = serverListener.streamListeners.get(0).headers;
+    assertEquals(userAgent, receivedHeaders.get(userAgentKey));
+  }
+
   /**
    * Verifies that we can create multiple TLS client transports from the same builder.
    */
   @Test
   public void creatingMultipleTlsTransportsShouldSucceed() throws Exception {
-    // Create the protocol negotiator.
-    File clientCert = TestUtils.loadCert("ca.pem");
-    SslContext clientContext = GrpcSslContexts.forClient().trustManager(clientCert).build();
-    ProtocolNegotiator negotiator = ProtocolNegotiators.tls(clientContext, address);
-
     // Create a couple client transports.
     for (int index = 0; index < 2; ++index) {
-      NettyClientTransport transport = new NettyClientTransport(address, NioSocketChannel.class,
-              group, negotiator, DEFAULT_WINDOW_SIZE, DEFAULT_WINDOW_SIZE);
-      transports.add(transport);
+      NettyClientTransport transport = createTransport();
```

Comment:
This change completely breaks the usefulness of this test. We need the same ProtocolNegotiator used for each transport.